crates/subspace-farmer/src/plot.rs

```diff
@@ -529,6 +622,32 @@ impl IndexHashToOffsetDB {
         }
     }
 
+    fn iter_from<'a>(
```

Comment:
I don't see why we need this smart thing that produces an iterator if we collect the result into vector anyway. That we then turn back into iterator and map again. All that should just be one simpler function.

Reply:
True, I'll remove iterator and just use simple function then

Reply:
Can I keep this iterator as it more flexible? Code fetching some specific amount of piece offsets would look the same, I think

Reply:
I just feel like you're creating more abstractions than necessary sometimes that makes codebase more complex without clear benefit to it rather than making code look "smart".

Reply:
Ikr, but here it won't benefit at all rewriting that without iterators. It would look nearly the same, I bet. I can rewrite it in such a manner for a clear comparison.

The thing which is distinct is that iterators are very flexible and can be reused.

Reply:
It is helpful when the code needs to be reused. If it doesn't need to be, then ability to do `?` makes code much more readable than composing `impl Iterator<Item = io::Result<PieceOffset>> + 'a`, collecting it to `io::Result<Vec<_>>` just to call `?` and turning back into iterator to do mapping. I suspect all that can be more readable when inlined and iterator removed with just a simple loop.